Tests/SwiftSyntaxMacroExpansionTest/MacroReplacementTests.swift

```diff
@@ -113,4 +114,230 @@ final class MacroReplacementTests: XCTestCase {
       """
     )
   }
+
+  func testMacroGenericArgumentExpansion_base() throws {
+    let macro: DeclSyntax =
+      """
+      macro gen<A, B>(a: A, b: B) = #otherMacro<A, B>(first: a, second: b)
+      """
+
+    let use: ExprSyntax =
+      """
+      #gen<Int, String>(a: 5, b: "Hello")
+      """
+
+    let macroDecl = macro.as(MacroDeclSyntax.self)!
+    let definition = try macroDecl.checkDefinition()
+    guard case let .expansion(expansion, replacements, genericReplacements) = definition else {
+      XCTFail("not a normal expansion")
+      return
+    }
+
+    let replacementA = try XCTUnwrap(genericReplacements.first)
+    let replacementB = try XCTUnwrap(genericReplacements.dropFirst().first)
+
+    XCTAssertEqual(genericReplacements.count, 2)
+
+    XCTAssertEqual(replacementA.parameterIndex, 0)
+    XCTAssertEqual("\(replacementA.reference.argument)", "A")
+
+    XCTAssertEqual(replacementB.parameterIndex, 1)
+    XCTAssertEqual("\(replacementB.reference.argument)", "B")
+
+    let expandedSyntax = macroDecl.expand(
+      use.as(MacroExpansionExprSyntax.self)!,
+      definition: expansion,
+      replacements: replacements,
+      genericReplacements: genericReplacements
+    )
+    assertStringsEqualWithDiff(
+      expandedSyntax.description,
+      """
+      #otherMacro<Int, String>(first: 5, second: "Hello")
+      """
+    )
+  }
+
+  func testMacroGenericArgumentExpansion_ignoreTrivia() throws {
+    let macro: DeclSyntax =
+      """
+      macro gen<A, B /* some comment */>(a: A, b: B) = #otherMacro<A, B>(first: a, second: b)
+      """
+
+    let use: ExprSyntax =
+      """
+      #gen<Int, String>(a: 5, b: "Hello")
+      """
+
+    let macroDecl = macro.as(MacroDeclSyntax.self)!
+    let definition = try macroDecl.checkDefinition()
+    guard case let .expansion(expansion, replacements, genericReplacements) = definition else {
+      XCTFail("not a normal expansion")
+      return
+    }
+
+    let replacementA = try XCTUnwrap(genericReplacements.first)
+    let replacementB = try XCTUnwrap(genericReplacements.dropFirst().first)
+    XCTAssertEqual(genericReplacements.count, 2)
+
+    XCTAssertEqual(replacementA.parameterIndex, 0)
+    XCTAssertEqual("\(replacementA.reference.argument)", "A")
+
+    XCTAssertEqual(replacementB.parameterIndex, 1)
+    XCTAssertEqual("\(replacementB.reference.argument)", "B")
+
+    let expandedSyntax = macroDecl.expand(
+      use.as(MacroExpansionExprSyntax.self)!,
```

Comment:
```suggestion
      use.cast(MacroExpansionExprSyntax.self),
```

Same a couple more times below